web/client/codechecker_client/thrift_call.py

```diff
@@ -21,6 +21,37 @@
 
 LOG = get_logger('system')
 
+PROXY_VALIDATION_NEEDED = True
+
+
+def proxy_settings_are_valid(transport):
```

Comment:
This function is not checking the `http[s]` scheme at the beginning, but line 73 says that it will be checked.
Minor thing: maybe the content of this function can go inside `validate_proxy_format()`, since it's quite short.